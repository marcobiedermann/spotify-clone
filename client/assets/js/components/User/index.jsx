import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

class User extends PureComponent {
  static propTypes = {
    display_name: PropTypes.string,
    followers: PropTypes.shape({
      total: PropTypes.number,
    }),
    images: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
      }),
    ),
  };

  static defaultProps = {
    display_name: '',
    followers: {
      total: 0,
    },
    images: [],
  };

  render() {
    const { display_name, followers, images } = this.props;

    return (
      <div>
        <h1>{display_name}</h1>
        {images[0] && (
          <figure>
            <img src={images[0].url} alt={display_name} />
          </figure>
        )}
        {followers && <p>{followers.total} Followers</p>}
      </div>
    );
  }
}

export default User;
