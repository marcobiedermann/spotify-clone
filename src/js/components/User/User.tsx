import React from 'react';
import Image from '../Image';

const defaultImageSize = 100;

interface Followers {
  total: number;
}

interface Image {
  height: number | null;
  url: string;
  width: number | null;
}

export interface UserProps {
  display_name: string;
  followers: Followers;
  images: Image[];
}

function User(props: UserProps): JSX.Element {
  const { display_name, followers, images } = props;
  const image = images[0];

  return (
    <div>
      <h1>{display_name}</h1>
      {image && (
        <figure>
          <Image
            {...image}
            width={image.width || defaultImageSize}
            height={image.height || defaultImageSize}
            alt={display_name}
          />
        </figure>
      )}
      <p>{followers.total} Followers</p>
    </div>
  );
}

export default User;
