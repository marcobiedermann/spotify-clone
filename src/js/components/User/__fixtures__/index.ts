const image = {
  height: 100,
  url: '',
  width: 100,
};

const user = {
  display_name: '',
  followers: {
    total: 100,
  },
  images: [image],
};

export default user;
