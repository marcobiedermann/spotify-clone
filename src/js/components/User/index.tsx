import React from 'react';
import Image from '../Image';

interface Followers {
  total: number;
}

interface Image {
  height: number;
  url: string;
  width: number;
}

export interface UserProps {
  display_name: string;
  followers?: Followers;
  images?: Image[];
}

function User(props: UserProps): JSX.Element {
  const { display_name, followers, images } = props;

  return (
    <div>
      <h1>{display_name}</h1>
      {images[0] && (
        <figure>
          <Image url={images[0].url} alt={display_name} />
        </figure>
      )}
      {followers && <p>{followers.total} Followers</p>}
    </div>
  );
}

export default User;
