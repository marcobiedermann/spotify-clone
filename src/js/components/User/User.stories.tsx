import type { Meta, StoryObj } from '@storybook/react';
import User from '.';
import userFixtures from './__fixtures__';

const meta: Meta<typeof User> = {
  component: User,
  args: userFixtures,
};

type Story = StoryObj<typeof User>;

const Default: Story = {};

export { Default };
export default meta;
