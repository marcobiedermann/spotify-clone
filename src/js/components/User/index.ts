import User from './User';

export default User;
export * from './User';
