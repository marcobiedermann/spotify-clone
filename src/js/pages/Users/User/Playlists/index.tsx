import React from 'react';
import { Helmet } from 'react-helmet';
import { useParams } from 'react-router-dom';
import { Error, Loader, Playlists } from '../../../../components';
import { useUsersPlaylists } from '../../../../hooks/users';

function PlaylistsPage(): JSX.Element {
  const { userId } = useParams();
  const { data, error, isError, isPending } = useUsersPlaylists(userId!);

  if (isError) {
    return <Error>{error.message}</Error>;
  }

  if (isPending) {
    return <Loader />;
  }

  const { items } = data;

  return (
    <>
      <Helmet>
        <title>Playlists</title>
      </Helmet>
      <Playlists items={items} />
    </>
  );
}

export default PlaylistsPage;
