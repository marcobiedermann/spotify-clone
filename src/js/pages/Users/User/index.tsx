import React from 'react';
import { Helmet } from 'react-helmet';
import { useParams } from 'react-router-dom';
import Error from '../../../components/Error';
import Loader from '../../../components/Loader';
import User from '../../../components/User';
import { useUser } from '../../../hooks/users';

function UserPage(): JSX.Element {
  const { userId } = useParams();
  const { data, error } = useUser(userId);

  if (error) {
    return <Error>{error.message}</Error>;
  }

  if (!data) {
    return <Loader />;
  }

  const { display_name } = data;

  return (
    <>
      <Helmet>
        <title>{display_name}</title>
      </Helmet>
      <User {...data} />
    </>
  );
}

export default UserPage;
