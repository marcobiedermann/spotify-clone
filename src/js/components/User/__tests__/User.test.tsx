import { render } from '@testing-library/react';
import React from 'react';
import User from '..';
import userFixtures from '../__fixtures__';

it('renders correctly', () => {
  const { container } = render(<User {...userFixtures} />);

  expect(container.firstChild).toMatchSnapshot();
});
