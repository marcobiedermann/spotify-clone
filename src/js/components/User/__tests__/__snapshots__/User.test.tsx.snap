// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1 />
  <figure>
    <img
      alt=""
      src=""
    />
  </figure>
  <p>
    100
     Followers
  </p>
</div>
`;
